internal/controller/components/modelregistry/modelregistry_support.go

```diff
@@ -68,3 +72,17 @@ func extraManifestInfo(sourcePath string) odhtypes.ManifestInfo {
 		SourcePath: path.Join(sourcePath, "extras"),
 	}
 }
+
+func ifGVKWatched(kvg schema.GroupVersionKind) func(context.Context, *odhtypes.ReconciliationRequest) bool {
+	return func(ctx context.Context, rr *odhtypes.ReconciliationRequest) bool {
+		if rr.DSCI.Spec.ServiceMesh != nil && rr.DSCI.Spec.ServiceMesh.ManagementState == operatorv1.Managed {
+			hasCRD, err := cluster.HasCRD(ctx, rr.Client, kvg)
+			if err != nil {
+				ctrl.Log.Error(err, "error checking if CRD installed", "GVK", kvg)
+				return false
+			}
+			return hasCRD
+		}
+		return false
+	}
+}
```

Comment:
The current function name `ifGVKWatched` seems misleading and doesn't accurately reflect what the function does. 

This function actually checks two specific conditions:
1. Whether `ServiceMesh` is defined and in a `Managed` state
2. Whether a specific CRD exists in the cluster

However, you can pass any unrelated `gvk`, i.e. `gvk.DSCinitialization`, and the function won't do what it is meant to do (IMHO). Given the function is only used with  `gvk.ServiceMeshMember` as an argument, maybe something like the code below could be simpler:

```suggestion
func isServiceMeshEnabled(ctx context.Context, rr *odhtypes.ReconciliationRequest) bool {
    // Check if ServiceMesh exists and is managed
    if rr.DSCI.Spec.ServiceMesh == nil || rr.DSCI.Spec.ServiceMesh.ManagementState != operatorv1.Managed {
        return false
    }
    
    // Check if the cluster has the `ServiceMeshMember` CRD installed
    hasCRD, err := cluster.HasCRD(ctx, rr.Client, gvk.ServiceMeshMember)
    if err != nil {
        ctrl.Log.Error(err, "error checking if CRD installed", "GVK", specificGVK)
        return false
    }
    
    return hasCRD
}
```